modules/nlu/src/backend/train-repo-router.ts

```diff
@@ -0,0 +1,91 @@
+import express, { Router } from 'express'
+import Joi from 'joi'
+import { NLUApplication } from './application'
+import { TrainingSession } from './application/typings'
+
+const trainingStatusSchema = Joi.object().keys({
+  status: Joi.string()
+    .valid(['idle', 'done', 'needs-training', 'training-pending', 'training', 'canceled', 'errored'])
+    .required(),
+  progress: Joi.number()
+    .default(0)
+    .min(0)
+    .max(1)
+    .required()
+})
+
+/**
+ * This is the training repository router to override the training states
+ *
+ * To get all trainings for the bot:
+ * GET /mod/nlu/trainrepo/trainings
+ *
+ * To get a training's state for a specific language:
+ * GET /mod/nlu/trainrepo/trainings/:lang
+ *
+ * To modify a training's state:
+ *
+ * POST /mod/nlu/trainrepo/trainings/:lang
+ * body: {
+ *   status: <One of 'idle', 'done', 'needs-training', 'training-pending', 'training', 'canceled', 'errored'>,
+ *   progress: <value between 0.0 and 1.0>
+ * }
+ *
+ * To delete a training state:
+ * POST /mod/nlu/trainrepo/trainings/:lang/delete
+ *
+ */
+const createRepositoryRouter = (app: NLUApplication): Router => {
+  const router = express.Router({ mergeParams: true })
```

Comment:
mmmhhhh, is this router protected by `Authorization` Header ?

would still use `bp.http.createRouterForBot`

Reply:
yes it is still protected, a child router request goes through all the middlewares of the parent!

Do you want me to create a router thats not under /mod/nlu? Because that's the only choice I have if I use the sdk.http.createRouterForBot

Reply:
allright!! just tested it and everything works fine.